usage.md

```diff
@@ -0,0 +1,11 @@
+# Usage
+
+This document provides an overview of the project, as well as documentation
+about the classes and other components contained within the project.
+
+## Types
+
+### `task<T>`
+
+Represents an operation which produces a value of type T.
+
```

Comment:
Any way to get automatically generated documentation that's as high-quality as the Rust stuff? Doxygen is _not_ fun to browse through, but I would like to have a lot of type-/method-level documentation in the source code. Makes it easier for me to browse; don't have to go to a browser while in an IDE.

Reply:
That's actually a really good question, and I found this! It looks really pretty!

Github link: <https://github.com/michaelrsweet/codedoc> 
Link to site: <https://codedoc.cc/>

Reply:
Are there examples of it being used for interactive C++ documentation? A-la ScalaDoc

Reply:
Update: I was mistaken. codedoc.cc and the github link I posted are actually two separate projects that happen to share the same name. One is a small project that's really small and has an outdated looking UI, but supports C++. The other one is a larger, modern project with a clean UI, but it doesn't automatically generate docs from code. It's just a glorified static site generator, and you still write all the docs as markdown. 

That being said, ccls will show you any comments written that are written immediately before the definition of a type, function, or variable! (I'm not sure if this is what you wanted, but I hope it is!)
![image](https://user-images.githubusercontent.com/36810712/128104186-8353c7f4-baf3-4a06-ab01-6b976df51997.png)

Reply:
That's half of it! The other half is explorable/searchable web documentation with type signatures. I'll ask some more C++ programmers I know and I'll report back. It's not too important for this PR, so don't stress about it.

Reply:
There's also <hdoc.io>, and that one actually seems to work really well (it's based on libclang so it understands modern C++), but it's proprietary. The free version only permits one project per account. See pricing here: <https://hdoc.io/pricing/>